File: Task/Associative-array-Iteration/C++/associative-array-iteration-2.cpp
std::map<std::string, int> myDict;
myDict["hello"] = 1;
myDict["world"] = 2;
myDict["!"] = 3;

// iterating over key-value pairs:
for (std::map<std::string, int>::iterator it = myDict.begin(); it != myDict.end(); ++it) {
    // the thing pointed to by the iterator is an std::pair<const std::string, int>&
    const std::string& key = it->first;
    int& value = it->second;
    std::cout << "key = " << key << ", value = " << value << std::endl;
}

File: Task/Associative-array-Iteration/Java/associative-array-iteration-2.java
Map<String, Integer> myDict = new HashMap<>();
myDict.put("hello", 1);
myDict.put("world", 2);
myDict.put("!", 3);

// iterating over key-value pairs:
myDict.forEach((k, v) -> {
    System.out.printf("key = %s, value = %s%n", k, v);
});

// iterating over keys:
myDict.keySet().forEach(k -> System.out.printf("key = %s%n", k));

// iterating over values:
myDict.values().forEach(v -> System.out.printf("value = %s%n", v));
